typewise_alert.test.py

```diff
@@ -3,9 +3,30 @@
 
 
 class TypewiseTest(unittest.TestCase):
-  def test_infers_breach_as_per_limits(self):
-    self.assertTrue(typewise_alert.infer_breach(20, 50, 100) == 'TOO_LOW')
+
+    def test_infers_breach_as_per_limits(self):
+        self.assertEqual(typewise_alert.infer_breach(-20, 0, 35), 'TOO_LOW')
+        self.assertEqual(typewise_alert.infer_breach(120, 0, 45), 'TOO_HIGH')
+        self.assertEqual(typewise_alert.infer_breach(20, 0, 40), 'NORMAL')
+
+    def test_checks_if_input_data_valid(self):
+        self.assertTrue(typewise_alert.is_input_valid('PASSIVE_COOLING', 120))
+        self.assertFalse(typewise_alert.is_input_valid(None, -20))
+        self.assertFalse(typewise_alert.is_input_valid(None, None))
+        self.assertFalse(typewise_alert.is_input_valid('MED_ACTIVE_COOLING', None))
+        self.assertFalse(typewise_alert.is_input_valid('ULTRA_COOLING', 120))
+
+    def test_checks_classify_temperature_breach_for_cooling_types(self):
+        self.assertTrue(typewise_alert.classify_temperature_breach('HI_ACTIVE_COOLING', 46) == 'TOO_HIGH')
+        self.assertTrue(typewise_alert.classify_temperature_breach('MED_ACTIVE_COOLING', 41) == 'TOO_HIGH')
+        self.assertTrue(typewise_alert.classify_temperature_breach('PASSIVE_COOLING', 36) == 'TOO_HIGH')
+
+    def test_check_and_alert_yields_apt_alerts_for_breaches(self):
+        self.assertEqual(typewise_alert.check_and_alert('email', {'cooling_type': 'HI_ACTIVE_COOLING'}, -20), 'EMAIL_SENT')
```

Comment:
These asserts state that check_and_alert converts the passed parameters into one return value string.
If you had to re-state its functionality in terms of data-conversion, what would you state?